### PR TITLE
Airlock Code Update + Tweaks

### PR DESCRIPTION
**Pull request by uraniummeltdown**
Cleaned up door/airlock code:
Removed frozen airlock overlays as it was only used in place (Clown Planet map) which got removed
Moved airlock types out of airlock.dm to their own file
Lowercased all door objects for consistency and renamed the default "poddoor" to blast door
Moved a few vars to the /door/ level
Gave most door assemblies names, vault and centcom/ele airlocks have their own assemblies
Screwdrivering an airlock displays a message, emagged airlocks have a different desc
Highsec airlocks can no longer be built with metal sheets, need 4 plasteel for them
Added sci airlock to metal recipes and rcd, hatches to rcd

:cl:
add: You can now create vault door assemblies with 4 plasteel sheets
add: Science airlocks have been added to metal recipes and the RCD
add: Airtight and Maintenance Hatch added to the RCD
tweak: Highsecurity airlock assemblies are built with 4 plasteel instead of 4 metal
tweak: All doors and doorlike things (firelock, airlock, windoor, blast door, shutters, spacepod door) run off the Environment Power Channel
tweak: Emagged airlocks have a different description on examining
tweak: Screwdrivering an airlock now displays a message. No more guessing if the panel is open or not.
tweak: Cult airlocks will keep their old name on being converted.
/:cl:

### CODE REVIEW

**Comment by taukausanake** [comment]
You know that frozen airlock thing if I'm thinking it correctly could be useful for telling if the other side of a door is cold/open to space. I could think a similar thing for hot door; I don't think we have anything of that kind of functionality and might be kind of useful if not also overlooked

**Comment by uraniummeltdown** [comment]
You can always implement it later. The current frozen airlock overlays assumed all airlocks looked like regular station airlocks so ended up looking stupid on airlocks like the high security one

File: code/game/machinery/doors/door.dm [hunk 1]
```diff
@@ -166,36 +181,35 @@
 		sleep(6)
 		open()
 		emagged = 1
+		if(istype(src, /obj/machinery/door/airlock))
```
Comment (tigercat2000) [inline code/game/machinery/doors/door.dm]: Why not just put this in a proc override?
```
/obj/machinery/door/airlock/emag_act(mob/user)
    . = ..()
    if(emagged)
        desc = "<span class='warning'>Its access panel is smoking slightly.</span>"

**Comment by uraniummeltdown** [comment]
DNM yet, going to make a few small changes

**Comment by uraniummeltdown** [comment]
Ready for review @tigercat2000 @Fox-McCloud

**Comment by ktoma36** [comment]
Wait does emaging a airlock display the message to just the person doing it or to everyone nearby?

**Comment by uraniummeltdown** [comment]
If you examine an airlock that has been emagged it displays a message in red text

**Pull request merged**